src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvesterServiceBean.java

```diff
@@ -380,6 +353,68 @@ private Long processRecord(DataverseRequest dataverseRequest, Logger hdLogger, P
 
         return harvestedDataset != null ? harvestedDataset.getId() : null;
     }
+    
+    File retrieveProprietaryDataverseMetadata (String remoteApiUrl) throws IOException {
+        InputStream in;
+        int responseCode = 0;
+        HttpURLConnection con = null;
+        
+        File tempMetadataFile = File.createTempFile("meta", ".tmp");
+        
+        try {
+            URL url = new URL(remoteApiUrl);
+
+            con = (HttpURLConnection) url.openConnection();
+            con.setRequestProperty("User-Agent", "DataverseHarvester/3.0");
+            responseCode = con.getResponseCode();
+        } catch (MalformedURLException mue) {
+            throw new IOException ("Bad API URL: "+remoteApiUrl);
+        } 
+        
+        if (responseCode == 200) {
+            in = con.getInputStream();
+            
+            FileOutputStream tempOut = new FileOutputStream(tempMetadataFile);
+            
+            int bufsize;
+            byte[] buffer = new byte[4 * 8192];
+
+            while ((bufsize = in.read(buffer)) != -1) {
+                tempOut.write(buffer, 0, bufsize);
+                tempOut.flush();
+            }
+
+            in.close();
+            tempOut.close();
+            return tempMetadataFile;
+        }
+
+        throw new IOException("Failed to download extended metadata.");
+  
+    }
+    
+    private void deleteHarvestedDatasetIfExists(String persistentIdentifier, Dataverse harvestingDataverse, DataverseRequest dataverseRequest, List<String> deletedIdentifiers, Logger hdLogger) {
+        Dataset dataset = datasetService.getDatasetByHarvestInfo(harvestingDataverse, persistentIdentifier);
+        if (dataset != null) {
+            datasetService.deleteHarvestedDataset(dataset, dataverseRequest, hdLogger);
+            // TODO: 
+            // check the status of that Delete - see if it actually succeeded
+            deletedIdentifiers.add(persistentIdentifier);
+            return;
+        }
+        hdLogger.info("No dataset found for " + persistentIdentifier + ", skipping delete. ");
+    }
+    
+    private static String getProprietaryDataverseMetadataURL(String baseURL, String identifier) {
+
+        baseURL = baseURL.replaceAll("/oai", "");
```

Comment:
If you don't expect the pattern to occur multiple times, you might want to avoid this. It's very slow compared to the other available options.